nemo/collections/llm/gpt/data/pre_training.py

```diff
@@ -133,7 +142,7 @@ def gpt_dataset_config(self) -> "GPTDatasetConfig":
         from megatron.core.datasets.gpt_dataset import GPTDatasetConfig
 
         return GPTDatasetConfig(
-            blend=[[str(self.path)], [1.0]],
+            blend=[[str(path) for path in self.paths], self.weights],
```

Comment:
If self.weights is None & path is a List, can't we just set all the weights to be equal?

Reply:
I tried that initially, but I ran into issues with non-None weights + [num_val_samples=None](https://github.com/NVIDIA/NeMo/blob/d2a86cea28c30678c1c4b1e57beb153ba88d9863/nemo/collections/llm/gpt/data/pre_training.py#L93)